### PR TITLE
Fix for reflink - build reflink for macos11

### PR DESCRIPTION
**Pull request by olsen232**
## Description

Building for macos10.9 (as we were doing previously) or equivalently pip installing it with the macos python version we have, ends up with a stub version of reflink that can't actually reflink anything. 

Fixes a test that was failing due to reflink being less powerful than shutil.copy (it can't overwrite existing files, so we have to make sure the old files are out of the way).

Adds tests that actually check if things are reflinked.

https://github.com/koordinates/kart/issues/876

- [x] Have you reviewed your own change?
- [x] Have you included test(s)?
- [x] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1231,3 +1231,68 @@ def _check_lfs_hashes(repo, expected_file_count=None):
...
+    import reflink
+
+    from kart import is_windows
+    from kart.lfs_util import get_hash_and_size_of_file, get_local_path_from_lfs_hash
+
+    clone_checker = shutil.which("clone_checker")
+    fienode = shutil.which("fienode")
+    has_checker = bool(clone_checker or fienode)
+
+    def _check_tile_is_reflinked(tile_path, repo, do_raise_skip=False):
```
Comment (rcoup) [inline tests/conftest.py:1258]: wrt `do_raise_skip`, why _wouldn't_ we always skip if the OS/infra doesn't support the test/feature? That's the pattern we've used elsewhere IIUC

File: tests/point_cloud/test_workingcopy.py [hunk 2]
```diff
@@ -1244,3 +1246,28 @@ def test_working_copy_conflicting_extension(cli_runner, data_archive, requires_p
         r = cli_runner.invoke(["status"])
         assert r.exit_code == INVALID_OPERATION
         assert "More than one tile found in working copy with the same name" in r.stderr
+
+
+def test_working_copy_reflink(cli_runner, data_archive, check_tile_is_reflinked):
+    if is_windows:
+        # Nothing to test, so let's mark test will show as "passed".
```
Comment (rcoup) [inline tests/point_cloud/test_workingcopy.py:1253]: same here...

**Review by rcoup — APPROVED** [review]
LGTM 👍

File: tests/conftest.py (hunk 1 above)
Comment (olsen232) [reply tests/conftest.py:1258]: I added a couple new tests, both called `test_working_copy_reflink`.
If reflink is missing, we skip them.
However, I also used this same fixture to add some more reflink asserts to some existing import tests. Those tests should now make extra asserts - if they are able to do so - but shouldn't skip the test if they can't.

File: tests/point_cloud/test_workingcopy.py (hunk 2 above)
Comment (olsen232) [reply tests/point_cloud/test_workingcopy.py:1253]: Yeah I was unsure what to do here: wasn't sure if I should try to use pytest.skip to mean "functionality is untested" and just a pass to mean "nothing to test". But I think you're right, pytest.skip makes more sense here.

**Pull request merged**